src/Plugin/QueueWorker/EloquaAppCloudQueueWorkerBase.php

```diff
@@ -0,0 +1,186 @@
+<?php
+
+namespace Drupal\eloqua_app_cloud\Plugin\QueueWorker;
+
+use Drupal\Core\Queue\QueueWorkerInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\Queue\QueueWorkerBase;
+use Psr\Log\LoggerInterface;
+
+
+/**
+ * Give it 60 seconds?
+ *
+ * @QueueWorker(
+ *  id = "eloqua_app_cloud_queue_worker_base",
+ *  title = @Translation("The Eloqua AppCloud Queue worker Base."),
+ *  cron = {"time" = 60},
+ * )
+ */
+abstract class EloquaAppCloudQueueWorkerBase extends QueueWorkerBase implements QueueWorkerInterface, ContainerFactoryPluginInterface {
+
+
+  /**
+   * {@inheritdoc}
+   */
+  public function processItem($queueItem) {
+
+  }
+
+  /**
+   * Wraps try/catch and retry attempts around Elomentary's $bulkApi->upload()
+   * function.
+   *
+   * @param {Object} $bulkApi
+   *    Elomentary bulkApi object reference.
+   *
+   * @param {Array} $contactsToBeStaged
+   *    Reference to the $contacts to be staged array.
+   *
+   * @param {Logger} $logger
+   *    Logger object for logging routine info / errors.
+   */
+  protected function tryBulkApiUpload($bulkApi, $contactsToBeStaged, LoggerInterface $logger) {
+    $retries = 1;
+    $maxRetries = 10;
+
+    try {
+      // Upload content to Eloqua staging area.
+      $bulkApi->upload($contactsToBeStaged);
+    } catch (Exception $e) {
+
+      $msg = "Upload Failed! Caught Exception: " . $e->getMessage();
+      $logger->error($msg);
+
+      // Retry for $maxRetries attempts.
+      for ($retries; $retries <= $maxRetries; $retries++) {
+        try {
+
+          // Upload content to Eloqua staging area.
+          $bulkApi->upload($contactsToBeStaged);
+          // Uploaded contacts successfully.
+          return;
+        } catch (Exception $e) {
+          $msg = "Reupload attempt #$retries Failed! Caught Exception: " . $e->getMessage();
+          $logger->error($msg);
+        }
+      }
+
+      $msg = "Upload failed after multiple retries.";
```

Comment:
Maybe worth throwing an exception here? Or somehow indicating failure so we can figure out what to do with these records (re-queue, etc).

Reply:
Thown.